src/firefly/js/visualize/task/PlotImageTask.js

```diff
@@ -398,3 +422,37 @@ function initBuildInDrawLayers() {
     DrawLayerCntlr.dispatchCreateDrawLayer(ActiveTarget.TYPE_ID);
 }
 
+/**
+ *
+ * @param {String[]} plotIdAry
+ * @param {Object.<string, OverlayPlotView>} oldOverlayPlotViews
+ */
+function matchAndActivateOverlayPlotViews(plotIdAry, oldOverlayPlotViews) {
+    plotIdAry.forEach( (plotId) => dispatchDeleteOverlayPlot({plotId, deleteAll:true}));
+
+    plotIdAry
+        .map( (plotId) => getPlotViewById(visRoot(), plotId))
+        .filter( (pv) => pv)
+        .forEach( (pv) => enableMatchingRelatedData(pv,oldOverlayPlotViews[pv.plotId]));
+}
+
+
+
+/**
+ *
+ * @param {String[]} plotIdAry
+ */
+function matchAndActivateOverlayPlotViewsByGroup(plotIdAry) {
+    const vr= visRoot();
+    plotIdAry
+        .map( (plotId) => getPlotViewById(visRoot(), plotId))
+        .filter( (pv) => pv)
+        .forEach( (pv) => {
+            const opvMatchArray= uniqBy(flatten(getPlotViewIdListInGroup(vr, pv.plotId)
+                                                       .filter( (id) => id!== pv.plotId)
+                                                       .map( (id) => getPlotViewById(vr,id))
+                                                       .map( (gpv) => gpv.overlayPlotViews)),
+                                   'maskNumber' );
```

Comment:
uniqBy can take a function or a string to use as the object key. In that case it will make sure the array of objects are unique by the key in each object. it is in the docs.
[http://devdocs.io/lodash~4/index#uniqBy](url)